### PR TITLE
1.1.15_wmagent branch keep only one job summary docs per agent and request

### PR DESCRIPTION
**Pull request by ticoann**
1.1.15_wmagent branch of #8836

### CODE REVIEW

**Comment by ticoann** [comment]
@amaltaro, Alan, test went well
I tested both fresh new agent and agent with old document. Old document gets deleted.

Please review.  You can either patch test agent for further test or one of the production agent.
When you patch this, please make sure both couch code and WMCore code are patched.
Coudh view need to be uploaded after patching the agent.

**Comment by amaltaro** [comment]
Thanks, Seangchan. I'll push it to all testbed nodes, tomorrow.

File: src/python/WMCore/Services/WMStats/WMStatsWriter.py [hunk 1]
```diff
@@ -77,6 +78,28 @@ def uploadData(self, docs):
             self.couchDB.queue(doc)
         return self.couchDB.commit(returndocs=True)
 
+    def bulkUpdateData(self, docs, existingDocs):
+        """
+        :param docs: docs to insert or update
+        :param existingDocs: docs existing in current
+        :return:
```
Comment (amaltaro) [inline src/python/WMCore/Services/WMStats/WMStatsWriter.py]: this line can go away :)

**Comment by amaltaro** [comment]
Thank you very much, Seangchan. I applied it to a couple of testbed agents and everything looks fine indeed. I'm going to push it to all the other testbed nodes now.

For the record, this is how we can apply this patch, in 5 steps:
```
curl https://patch-diff.githubusercontent.com/raw/dmwm/WMCore/pull/8840.patch | patch -d /data/srv/wmagent/current/apps/wmagent/lib/python2.7/site-packages/ -p3

curl https://patch-diff.githubusercontent.com/raw/dmwm/WMCore/pull/8840.patch | patch -d /data/srv/wmagent/current/apps/wmagent/data -p2

$manage stop-agent

$manage execute-agent wmagent-couchapp-init

$manage start-agent
```

**Comment by amaltaro** [comment]
Everything seems to be working fine. I made this deployment PR https://github.com/dmwm/deployment/pull/678 and Lina has applied it to the cmsweb-testbed backend running that thread. 

I still see my test wmstats documents, even though the agent is gone from the system.
I'll start preparing the same for production, but won't make any changes until you're back online, Seangchan.

**Comment by amaltaro** [comment]
Merging it to cut a new patch6 tag

**Comment by amaltaro** [comment]
[Fix #8840 - keep only one job summary docs per agent and request](https://trello.com/c/uT9f4hMM/129-fix-8840-keep-only-one-job-summary-docs-per-agent-and-request)

**Pull request merged**